x/nexus/migrations/v014/store.go

```diff
@@ -0,0 +1,188 @@
+	evmV013 "github.com/axelarnetwork/axelar-core/x/evm/migrations/v013"
+	evmtypes "github.com/axelarnetwork/axelar-core/x/evm/types"
+	"github.com/axelarnetwork/axelar-core/x/nexus/exported"
+)
+
+// MigrateStore performs in-place store migrations from v0.13 to v0.14. The
+// migration includes:
+// - migrate Assets from CosmosChain struct to ChainState
+// - migrate token MinAmount from TokenMetaDAta to ChainState
+func MigrateStore(ctx sdk.Context, storeKey sdk.StoreKey, cdc codec.BinaryCodec, axelarnetK types.AxelarnetKeeper, evmK types.EVMKeeper) error {
+	nexusStore := getStore(ctx.KVStore(storeKey), cdc)
+
+	if err := migrateChainState(nexusStore); err != nil {
+		return err
+	}
+
+	if err := migrateCosmosChainAssets(ctx, nexusStore, axelarnetK); err != nil {
+		return err
+	}
+
+	if err := migrateEvmAssets(ctx, cdc, nexusStore, evmK); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func migrateChainState(store utils.KVStore) error {
+	for _, chain := range v013.GetChains(store) {
+		oldChainState, ok := v013.GetChainState(store, chain)
+		if !ok {
+			return fmt.Errorf("failed to get chain state %s", chain.Name)
+		}
+
+		newChainState := types.ChainState{
+			Chain: exported.Chain{
+				Name:                  chain.Name,
+				SupportsForeignAssets: chain.SupportsForeignAssets,
+				KeyType:               chain.KeyType,
+				Module:                chain.Module,
+			},
+			Maintainers: oldChainState.Maintainers,
+			Activated:   oldChainState.Activated,
+		}
+		deleteChainState(store, chain.Name)
+		setChainState(store, newChainState)
+	}
+
+	return nil
+}
+
+func migrateCosmosChainAssets(ctx sdk.Context, nexusStore utils.KVStore, axelarnetK types.AxelarnetKeeper) error {
+	// get all cosmos chains
+	for _, chainName := range axelarnetK.GetCosmosChains(ctx) {
+		cosmosChain, ok := axelarnetK.GetCosmosChainByName(ctx, chainName)
+		if !ok {
+			return fmt.Errorf("failed to find cosmos chain %s", cosmosChain.Name)
+		}
+
+		chain, ok := getChain(nexusStore, cosmosChain.Name)
+		if !ok {
+			return fmt.Errorf("failed to find chain %s", cosmosChain.Name)
+		}
+
+		for _, asset := range cosmosChain.Assets {
+			// register asset as native asset in chain state
+			err := registerAsset(nexusStore, chain, exported.NewAsset(asset.Denom, asset.MinAmount, true))
+			if err != nil {
+				return err
+			}
+
+			if chain.Name != axelarnet.Axelarnet.Name {
+				err = registerAsset(nexusStore, axelarnet.Axelarnet, exported.NewAsset(asset.Denom, asset.MinAmount, false))
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func migrateEvmAssets(ctx sdk.Context, cdc codec.BinaryCodec, nexusStore utils.KVStore, evmK types.EVMKeeper) error {
+
+	for _, chain := range getChains(nexusStore) {
+		if chain.Module != evmtypes.ModuleName {
+			continue
+		}
+
+		// get chain keeper
+		chainK := evmK.ForChain(chain.Name)
+
+		for _, token := range chainK.GetTokens(ctx) {
+			token.
+		}
```

Comment:
i assume these lines should be deleted?

Reply:
I am doing another change, will let you when I finish. thanks for your input so far!